### PR TITLE
feat(widget-cfg): update web3modal

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

Updated [`web3modal`](https://web3modal.com/) in the widget configurator to the latest version.

# To Test

Wallet connection should work as before

### CODE REVIEW

File: apps/widget-configurator/src/web3modalConfig.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+const ethersConfig = defaultConfig({
+  metadata,
+
+  /*Optional*/
+  enableEIP6963: true,
+  enableInjected: true,
+  enableCoinbase: true, // true by default
+  rpcUrl: '...', // used for the Coinbase SDK
```
Comment (alfetopito) [inline apps/widget-configurator/src/web3modalConfig.ts:25]: Don't we need to set this, give that coinbase is enabled?

File: apps/widget-configurator/src/app/configurator/hooks/useProvider.ts [hunk 2]
```diff
@@ -1,49 +1,9 @@
...
 import { EthereumProvider } from '@cowprotocol/widget-lib'
 
-import { useAccount, useConfig } from 'wagmi'
-
-import EventEmitter from 'events'
-
-const onDisconnect = new EventEmitter()
+import { useWeb3ModalProvider } from '@web3modal/ethers5/react'
 
 export function useProvider(): EthereumProvider | undefined {
```
Comment (alfetopito) [inline apps/widget-configurator/src/app/configurator/hooks/useProvider.ts:5]: Wooooow so much less code ❤️

**Comment by elena-zh** [comment]
Hey @shoom3301 , the build failed. Could you please fox this?

File: apps/widget-configurator/src/web3modalConfig.ts (hunk 1 above)
Comment (shoom3301) [reply apps/widget-configurator/src/web3modalConfig.ts]: Fixed

File: apps/widget-configurator/src/web3modalConfig.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+      return {
+        chainId,
+        name: info.label,
+        rpcUrl: RPC_URLS[chainId],
+        explorerUrl: info.explorer,
+        currency: info.nativeCurrency.symbol || '',
+      }
+    }),
+    projectId,
+    enableAnalytics: true, // Optional - defaults to your Cloud configuration
```
[the remark this answers is not in the document]
Comment (shoom3301) [reply apps/widget-configurator/src/web3modalConfig.ts]: https://docs.walletconnect.com/web3modal/react-native/options#enableanalytics

Anyway, disabled it

**Comment by shoom3301** [comment]
@elena-zh fixed

**Review by elena-zh — CHANGES_REQUESTED** [review]
Hey @shoom3301 , great!
However, it would be great to change infura to nodereal requests for Ethereum and Sepolia networks
infura
I assume, due to this issue I've not been able to wrap
wrap
And from the UI perspective now the connection info exceeds the configurator area 
huge

Could you please fix these issues?

**Comment by shoom3301** [comment]
@elena-zh could you check it again please? I don't observe it anymore

**Review by anxolin — APPROVED** [review]

**Comment by elena-zh** [comment]
Hey @shoom3301 , great! Thanks for fixing the network switch issue: now it works fine. 
As for the icon issue, I noticed that it starts looking better when uncheck this box in the console: 
image
Is this possible to save this? :)

**Pull request merged**